### DISCUSSION TITLE
Allow document properties to be marked as Sensitive Data elements.

### DISCUSSION THREAD

**Discussion by wicketbr** [umbraco/Umbraco-CMS#17221] [Features and ideas]
I would like to be able to make certain properties for document types as Sensitive Data elements so that they are only visible to members of the "Sensitive data" group. As it is currently it appears that the Sensitive Data option is only available for Member Types, but that feature would also be useful to make fields that can only be edited by certain users.

**Comment by nul800sebastiaan** [comment]
Thanks for the feature idea @wicketbr!

There's the Admin Only Property Editor package (not for v14+ yet)  that could be used for this: https://marketplace.umbraco.com/package/our.umbraco.community.adminonlyproperty

However, I will list some problems I foresee with hiding property data for certain users:

- What happens when someone who can't see the property publishes the document, maybe someone was working on that property and it was not ready to be published yet, hard to tell if you can't see it
- This data is going to get published at some point, so now potentially it can be seen in "public"
  - For example: enabling the Headless API, you'd have an API endpoint that would contain that data
- Maybe there's an RSS feed that contains this property publicly
- Maybe there's a custom API that exposes publicly

Ultimately, you are working on content that is going to be published, for the public internet (or for logged in members). That is not a great place to store sensitive data. 

This is why we haven't built this as a feature so far.

Long post, sorry, but what I really came here to ask is: what is your use case? What do those certain users need to edit that other users in the backoffice shouldn't be seeing?

Thanks again, look forward to hearing back.

**Comment by wicketbr** [comment]
I'm thinking of properties that are used for admin purposes. 

Things like a Google Analytics code, or settings related to securing content or what layout might be used for a given page. Things that an editor shouldn't see and that don't directly relate to the published content. 

I have options for whether or not an item appears in the navigation menu, items related to whether or not a page is secured. 

Also, I'd like to be able to have an area where we can select from and insert blocks used for admin purposes only. 
We do a lot of custom development in the platform and I have some admin blocks that are used to insert code solutions that I wouldn't want my regular editors to be able to select. 

We also have fields for adding javascript, css, references to external javascript and css files. 

There have also been times where we have added fields to turn on or off features that we want to manage at an administrative level and not have the users be able to turn those features on and off. 

I'm sure there are other examples people would have of items they might want to allow only a specific group of users to be able to edit. 

The admin setting up the content would be responsible for making sure they are not hiding items related to published content that might cause conflicts.

**Reply by nul800sebastiaan** [reply]
> Things like a Google Analytics code, or settings related to securing content or what layout might be used for a given page. 

What most people do for things like this is they have a settings node in the root of their content tree where "things" like this get configured. The node lives outside the tree where editors can work, so:

- Content `<-- admins start node is here`
  - Home `<-- editors start node is here`
  - Settings `<-- editors can't see nor access this node`

> Also, I'd like to be able to have an area where we can select from and insert blocks used for admin purposes only.  

What is kinds of blocks are theywhat does it do?

> We also have fields for adding javascript, css, references to external javascript and css files. 

I don't quite understand why these need to be managed on a content item level, instead of in the template where I would expect them to go?

**Comment by wicketbr** [comment]
We have about 100 different sites that all share the same layout template, so we use these other properties to do customization for individual sites. Our university template is used on all the sites, but there might be slight tweaks needed for a given user that can't be done in the template, since the template is shared between all the sites.
